src/fuse_opts.rs

```diff
@@ -0,0 +1,134 @@
+
+
+//
+// FUSE arguments (see fuse_opt.h for details)
+//
+
+use libc::{c_char, c_int};
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct fuse_args {
+    pub argc: c_int,
+    pub argv: *const *const c_char,
+    pub allocated: c_int,
+}
+
+cfg_if! {
+    if #[cfg(feature="rust-mount")] {
+
+        use libc::{mode_t, uid_t, gid_t};
+        use getopts::Options;
+        use std::str::FromStr;
+        use std::ffi::CStr;
+        use std::slice;
+
+        #[derive(Copy, Clone, Debug, PartialEq)]
+        pub struct ParseError;
+
+        pub enum MetaFuseOpt {
+            Uid(uid_t),
+            Gid(gid_t),
+            RootMode(mode_t),
+            AllowOther,
+            DefaultPermissions,
+            NonEmpty,
+            SubType(String),
+            #[cfg(target_os = "darwin")]
+            Volname(String), // osxfuse option
+        }
+
+        impl FromStr for MetaFuseOpt {
+            type Err = ParseError;
+
+            fn from_str(s :&str) -> Result<Self, Self::Err> {
+                let (a, b) = s.split_at(s.find("=").unwrap_or(s.len()));
+
+                match (a, String::from(b.trim_left_matches('='))) {
+                    ("user_id", val) => Ok(MetaFuseOpt::Uid(val.parse::<uid_t>().unwrap())),
+                    ("group_id", val) => Ok(MetaFuseOpt::Gid(val.parse::<gid_t>().unwrap())),
+                    ("rootmode", val) => Ok(MetaFuseOpt::RootMode(val.parse::<mode_t>().unwrap())),
+                    ("allow_other", _) => Ok(MetaFuseOpt::AllowOther),
+                    ("default_permissions", _) => Ok(MetaFuseOpt::DefaultPermissions),
+                    ("nonempty", _) => Ok(MetaFuseOpt::NonEmpty),
+                    ("subtype", val) => Ok(MetaFuseOpt::SubType(val.clone())),
+                    #[cfg(target_os = "darwin")]
+                    ("volname", val) => Ok(MetaFuseOpt::Volname(val.clone())),
+                    _ => Err(ParseError),
+                }
+            }
+        }
+
+        impl ToString for MetaFuseOpt {
+            fn to_string(&self) -> String {
+                match *self {
+
+                    MetaFuseOpt::Uid(val) => format!("user_id={}", val),
+                    MetaFuseOpt::Gid(val) => format!("group_id={}", val),
+                    MetaFuseOpt::RootMode(val) => format!("rootmode={}", val),
+                    MetaFuseOpt::AllowOther => String::from("allow_other"),
+                    MetaFuseOpt::DefaultPermissions => String::from("default_permissions"),
+                    MetaFuseOpt::NonEmpty => String::from("nonempty"),
+                    MetaFuseOpt::SubType(ref val) => format!("subtype={}", val),
+                    #[cfg(target_os = "darwin")]
+                    MetaFuseOpt::Volname(ref val) => format!("volname={}", val),
+                }
+            }
+        }
+
+        //
+        // fn get_get(type: ident, tab: FuseOpts) -> Option<ident::value>
+        // 
+        macro_rules! get_opt {
```

Comment:
What does this do ? Why is it a macro and not a function ?

Reply:
Because I need to use ident MetaFuseOpt only implemented in macro as key for look in my vector

Reply:
Ah, I see now. That's... sort of icky xD